From 625db17095325b37dc64e5e95d6b72883d981d68 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Sat, 29 May 2010 09:44:53 +1000
Subject: [BUILTIN] Continue after EINTR in read(1) with no pending signals

The recent introduction of SIGCHLD trapping broke read(1) as
each SIGCHLD may cause read(1) to return prematurely.  Now if
we did have a trap for SIGCHLD read(1) should actually do this.
However, returning when SIGCHLD isn't trapped is wrong.

This patch fixes this by checking for EINTR and pendingsigs in
read(1).

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit 0cda2e1f8d8222fa497f808b54a2146d60e304f5)

Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 ChangeLog |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 5b0bb5a..4769054 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2010-05-29  Herbert Xu <herbert@gondor.apana.org.au>
+
+	* Continue after EINTR in read(1) with no pending signals.
+
 2010-05-27  Jilles Tjoelker <jilles@stack.nl>
 
 	* Force fork if any trap is set, not just on EXIT.
-- 
1.7.4.1

